code/go/0chain.net/core/lock/lock.go

```diff
@@ -1,17 +1,70 @@
 package lock
 
-import "sync"
+import (
+	"sync"
+	"time"
+)
 
-var lockPool = make(map[string]*sync.Mutex, 0) //nolint:gosimple // need more time to verify: declaring capacity is probably necessary?
-var lockMutex = &sync.Mutex{}
+var (
+	lockPool   = make(map[string]*Mutex)
+	unlockPool = make(map[string]time.Time)
+	lockMutex  sync.Mutex
+)
 
-func GetMutex(tablename string, key string) *sync.Mutex {
+// Mutex a mutual exclusion lock.
+type Mutex struct {
+	// key lock key in pool
+	key string
+	sync.Mutex
+}
+
+// Lock implements Locker.Lock
+func (m *Mutex) Lock() {
+	m.Mutex.Lock()
+}
+
+// Unlock implements Locker.Unlock, and mark mutex as unlock object
+func (m *Mutex) Unlock() {
+	lockMutex.Lock()
+	defer lockMutex.Unlock()
+
+	m.Mutex.Unlock()
+	//mark it as unlock object, it will be deleted in clean worker
+	unlockPool[m.key] = time.Now()
+}
+
+// GetMutex get mutex by table and key
+func GetMutex(tablename string, key string) *Mutex {
 	lockKey := tablename + ":" + key
 	lockMutex.Lock()
 	defer lockMutex.Unlock()
 	if eLock, ok := lockPool[lockKey]; ok {
+		// do NOT remove it from pool
+		delete(unlockPool, lockKey)
 		return eLock
 	}
-	lockPool[lockKey] = &sync.Mutex{}
-	return lockPool[lockKey]
+
+	m := &Mutex{key: lockKey}
+	lockPool[lockKey] = m
+
+	return m
+}
+
+func init() {
+	go startLockPoolCleaner()
+}
+
+func startLockPoolCleaner() {
```

Comment:
Personally, I would prefer to use a reference counting method, you could use [SetFinalizer](https://pkg.go.dev/runtime#SetFinalizer) to control the count.

However, if using this method I think it might be best to control the loop from blobber.main, perhaps add to the other [worker threads](https://github.com/0chain/blobber/blob/master/code/go/0chain.net/blobber/main.go#L179).